arch/arm/src/gd32f4/gd32f4xx_sdio.c

```diff
@@ -0,0 +1,3133 @@
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_SDIO_XFRDEBUG
+static void gd32_sample(struct gd32_dev_s *priv, int index)
+{
+  struct gd32_sampleregs_s *regs = &g_sampleregs[index];
+
+#if defined(CONFIG_DEBUG_DMA_INFO) && defined(CONFIG_GD32F4_SDIO_DMA)
+  if (priv->dmamode)
+    {
+      gd32_dma_sample(priv->dma, &regs->dma);
+    }
+#endif
+
+  gd32_sdiosample(&regs->sdio);
+}
+#endif
+
+/****************************************************************************
+ * Name: gd32_sdiodump
+ *
+ * Description:
+ *   Dump one register sample
+ *
+ * Input Parameters:
+ *   regs - SDIO registers structure
+ *   msg  - message data
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_SDIO_XFRDEBUG
+static void gd32_sdiodump(struct gd32_sdioregs_s *regs, const char *msg)
+{
+  mcinfo("SDIO Registers: %s\n", msg);
+  mcinfo("  POWER[%08x]: %08x\n", GD32_SDIO_PWRCTL,  regs->power);
+  mcinfo(" CLKCTL[%08x]: %08x\n", GD32_SDIO_CLKCTL,  regs->clkctl);
+  mcinfo("DATACTL[%08x]: %08x\n", GD32_SDIO_DATACTL, regs->datactl);
+  mcinfo(" DATATO[%08x]: %08x\n", GD32_SDIO_DATATO,  regs->datato);
+  mcinfo("DATALEN[%08x]: %08x\n", GD32_SDIO_DATALEN, regs->datalen);
+  mcinfo(" DCOUNT[%08x]: %08x\n", GD32_SDIO_DATACNT, regs->datacnt);
+  mcinfo("   STAT[%08x]: %08x\n", GD32_SDIO_STAT,    regs->stat);
+  mcinfo("  INTEN[%08x]: %08x\n", GD32_SDIO_INTEN,   regs->inten);
+  mcinfo("FIFOCNT[%08x]: %08x\n", GD32_SDIO_FIFOCNT, regs->fifocnt);
+}
+#endif
+
+/****************************************************************************
+ * Name: gd32_dumpsample
+ *
+ * Description:
+ *   Dump one register sample
+ *
+ * Input Parameters:
+ *   priv - A reference to the SDIO device state structure
+ *   regs - SDIO registers structure
+ *   msg  - message data
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_SDIO_XFRDEBUG
+static void gd32_dumpsample(struct gd32_dev_s *priv,
+                             struct gd32_sampleregs_s *regs,
+                             const char *msg)
+{
+#if defined(CONFIG_DEBUG_DMA_INFO) && defined(CONFIG_GD32F4_SDIO_DMA)
+  if (priv->dmamode)
+    {
+      gd32_dma_sample (priv->dma, &regs->dma, msg);
+    }
+#endif
+
+  gd32_sdiodump(&regs->sdio, msg);
+}
+#endif
+
+/****************************************************************************
+ * Name: gd32_dumpsamples
+ *
+ * Description:
+ *   Dump all sampled register data
+ *
+ * Input Parameters:
+ *   priv - A reference to the SDIO device state structure
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_SDIO_XFRDEBUG
+static void gd32_dumpsamples(struct gd32_dev_s *priv)
+{
+  gd32_dumpsample(priv, &g_sampleregs[SAMPLENDX_BEFORE_SETUP],
+                   "Before setup");
```

Comment:
```suggestion
  gd32_dumpsample(priv, &g_sampleregs[SAMPLENDX_BEFORE_SETUP],
                  "Before setup");
```
same in other similar lines below